### ISSUE TITLE
Give the user the ability to specify which custom keywords accept other keywords

### ISSUE THREAD

**Issue by Serhiy1** [robocorp/robotframework-lsp#821]
**Is your feature request related to a problem? Please describe.**
I have a keyword that uses polls using custom logic

``` robot
Wait For
    [Arguments]    ${keyword}    @{args}    ${poll_time}=1 minute    &{kwargs}
    [Documentation]    Geometric back off until keyword is successful or sleep values run out

``` 

When I use `Wait For`, the `${keyword}` is always interpreted as a string. This makes it challenging to navigate my suites as I need to do global searches. 

**Describe the solution you'd like**
I want to tell LSP that the `wait for` keyword accepts another keyword for its first argument and IntelliSense (go to definition) to work for the passed-in keyword.

**Comment by fabioz** [comment]
Maybe the language server could identify something in the documentation to be able to recognize that...

How about something as:

`HINT:KEYWORDARG:${keyword}`

**Comment by Serhiy1** [comment]
Yeah, that's pretty easy to do on my side. Though parsing docstrings seems complicated from the LSP maintainer side.

I also don't mind setting values and settings.json  

something like 
```
"robot.lint.acceptsKeywordArg":
    {
         "Wait For" : ["${keyword}"] 
    }
```
